File: dotnet/Logify.Alert.Core/Collectors/LogifyAppInfoCollector.cs
using System;

namespace DevExpress.Logify.Core.Internal {
    public interface ILogifyAppInfo {
        string AppName { get; }
        string AppVersion { get; }
        string UserId { get; }
    }
    public class LogifyAppInfoCollector : IInfoCollector {
        readonly ILogifyAppInfo appInfo;

        public LogifyAppInfoCollector(ILogifyAppInfo appInfo) {
            this.appInfo = appInfo;
        }

        public string AppName { get; set; }
        public string AppVersion { get; set; }
        public string UserId { get; set; }

        public virtual void Process(Exception ex, ILogger logger) {
            logger.BeginWriteObject("logifyApp");
            try {
                logger.WriteValue("name", String.IsNullOrEmpty(AppName) ? appInfo.AppName : AppName);
                logger.WriteValue("version", String.IsNullOrEmpty(AppVersion) ? appInfo.AppVersion : AppVersion);
                logger.WriteValue("userId", String.IsNullOrEmpty(UserId) ? appInfo.UserId : UserId);
            }
            finally {
                logger.EndWriteObject("logifyApp");
            }
        }
    }
}

File: java/Core/src/main/java/com/devexpress/logify/alert/core/collectors/LogifyAppInfoCollector.java
package com.devexpress.logify.alert.core.collectors;

import com.devexpress.logify.alert.core.ApplicationProperties;
import com.devexpress.logify.alert.core.logger.ILogger;

public class LogifyAppInfoCollector implements IInfoCollector {

    protected ApplicationProperties applicationProperties;

    public LogifyAppInfoCollector(ApplicationProperties applicationProperties){
        this.applicationProperties = applicationProperties;
    }

    protected ApplicationProperties getApplicationProperties(){
        return this.applicationProperties == null ? new ApplicationProperties() : this.applicationProperties;
    }
    protected String getAppName() {
        return getApplicationProperties().AppName;
    }

    protected String getAppVersion() {
        return getApplicationProperties().AppVersion;
    }

    protected String getUserId() {
        return getApplicationProperties().UserId;
    }

    public void process(Throwable ex, ILogger logger) {
        logger.beginWriteObject("logifyApp");
        try {
            String name = getAppName();
            if (name != null)
                logger.writeValue("name", name);
            String version = getAppVersion();
            if (version != null)
                logger.writeValue("version", version);
            String userId = getUserId();
            if (userId != null)
                logger.writeValue("userId", userId);
        } finally {
            logger.endWriteObject("logifyApp");
        }
    }
}